src/App.tsx

```diff
@@ -1,14 +1,87 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getActiveTodos, getcompletedTodos, getTodos, getUser } from './api';
+import { Todo, FilterTypes } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [preparedTodos, setPreparedTodos] = useState<Todo[]>([]);
+  const [filterStatus, setFilterStatus] = useState<FilterTypes>(
+    FilterTypes.All,
+  );
+  const [textInput, setTextInput] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [todoModalStatus, setTodoModalStatus] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [preparedUser, setPreparedUser] = useState<User | null>(null);
+
+  const handleSelectFilter = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilterStatus(event.target.value as FilterTypes);
+  };
+
+  const handleTextInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTextInput(event.target.value);
+  };
+
+  const handleRemoveFilter = () => {
+    setFilterStatus(FilterTypes.All);
+    setTextInput('');
+  };
+
+  const handleOpenModal = (todo: Todo) => {
+    setSelectedTodo(todo);
+    setTodoModalStatus(true);
+  };
+
+  const handleCloseModal = () => {
+    setTodoModalStatus(false);
+  };
+
+  useEffect(() => {
+    let todos: Promise<Todo[]>;
+
+    setIsLoading(true);
+
+    if (FilterTypes.All === filterStatus) {
+      todos = getTodos();
+    } else if (FilterTypes.Active === filterStatus) {
+      todos = getActiveTodos();
+    } else if (FilterTypes.Completed === filterStatus) {
+      todos = getcompletedTodos();
+    }
+
+    todos
+      .then(data => {
+        const filtered = data.filter(todo =>
+          todo.title
+            .toLowerCase()
+            .trim()
+            .includes(textInput.toLowerCase().trim()),
+        );
+
+        setPreparedTodos(filtered);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [filterStatus, textInput]);
```

Comment:
AI style
why we need useEffect? can we just load todo in useEffect and have preparedTodos function?
```suggestion
  useEffect(() => {
    todos
      .then(data => {
        const filtered = data.filter(todo =>
          todo.title
            .toLowerCase()
            .trim()
            .includes(textInput.toLowerCase().trim()),
        );

        setPreparedTodos(filtered);
      })
      .finally(() => {
        setIsLoading(false);
      });
  }, [filterStatus, textInput]);
```